rest_framework/relations.py

```diff
@@ -169,6 +169,12 @@ def get_queryset(self):
             # for the field.
             # Eg: 'MyRelationship(queryset=ExampleModel.objects.all())'
             queryset = queryset.all()
+        if hasattr(self.parent, "Meta") and hasattr(self.parent.Meta, "model"):
```

Comment:
I'm not keen on this way around, as it feels like it'll end up being brittle, and also wouldn't work with plain serializer classes. In other cases, rather than introspect inside a relationship if it belongs to a modelserializer, we *change* what kind of field gets automatically generated for `ModelSerializer`.

So...

I *think* what we want is for `ModelSerializer.get_fields` to be returning a relational field that has the `queryset` argument already filtered.

This calls into `build_field()`, which for this case calls into `build_relational_field()`, which calls `get_relation_kwargs`. So...

I *think* what we want is for this `get_relation_kwargs ` function... https://github.com/encode/django-rest-framework/blob/master/rest_framework/utils/field_mapping.py#L236

To filter the `queryset` keyword argument appropriately, if `get_limit_choices_to()` returns something.